### ISSUE TITLE
[Bug]: There are some deadlock detected error during stability test  in tpcc and sysbench mixed cases after cn was crash 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#13780]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.1-dev, main, 1.0

### Commit ID

981aa996ae06a3b5bbbaa7bfefe6654378e36a18

### Other Environment Information

```Markdown
- Hardware parameters:
  3*CN  16C  64G
  1*CN  16C  64G
  3*LOG 4C  16G
- OS type:Centos
- Others:
```

### Actual Behavior

 **during stability test  in tpcc and sysbench mixed, sometimes cn will be crash by oom, the lots of query will be error.such as:**
The last packet successfully received from the server was 60,060 milliseconds ago.  The last packet sent successfully to the server was 60,071 milliseconds ago.
2023-12-24 00:05:04 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20502, ErrorMessage : the connection between CN and TN has been disconnected
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:04 FATAL jTPCCTerminal:212 - [UNEXPECTED][TT_PAYMENT][EXECUTION] ErrorCode : 1105, ErrorMessage : dial tcp4 10.10.4.188:6003: connect: connection refused
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:04 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20502, ErrorMessage : the connection between CN and TN has been disconnected
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:04 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 1105, ErrorMessage : dial tcp4 10.10.4.188:6003: connect: connection refused
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:04 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 1105, ErrorMessage : dial tcp4 10.10.4.188:6003: connect: connection refused
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:04 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20502, ErrorMessage : the connection between CN and TN has been disconnected
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:26 FATAL jTPCCTerminal:294 - [UNEXPECTED][DELIVERY_BG][EXECUTION] ErrorCode : 20703, ErrorMessage : lock table not found on remote lock service
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:26 FATAL jTPCCTerminal:212 - [UNEXPECTED][TT_PAYMENT][EXECUTION] ErrorCode : 20703, ErrorMessage : lock table not found on remote lock service
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:28 FATAL jTPCCTerminal:212 - [UNEXPECTED][TT_PAYMENT][EXECUTION] ErrorCode : 20703, ErrorMessage : lock table not found on remote lock service
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:28 FATAL jTPCCTerminal:294 - [UNEXPECTED][DELIVERY_BG][EXECUTION] ErrorCode : 20703, ErrorMessage : lock table not found on remote lock service

**but nearly every time in this period, some deadlock detected error will occure. e.g.** 
2023-12-24 00:05:28 FATAL jTPCCTerminal:212 - [UNEXPECTED][TT_PAYMENT][EXECUTION] ErrorCode : 20703, ErrorMessage : lock table not found on remote lock service
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:28 FATAL jTPCCTerminal:294 - [UNEXPECTED][DELIVERY_BG][EXECUTION] ErrorCode : 20703, ErrorMessage : lock table not found on remote lock service
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:28 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20702, ErrorMessage : lock table bind changed
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:31 FATAL jTPCCTerminal:325 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20702, ErrorMessage : lock table bind changed
2023-12-24 00:05:31 FATAL jTPCCTerminal:294 - [UNEXPECTED][DELIVERY_BG][EXECUTION] ErrorCode : 20701, ErrorMessage : deadlock detected
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
2023-12-24 00:05:31 FATAL jTPCCTerminal:212 - [UNEXPECTED][TT_PAYMENT][EXECUTION] ErrorCode : 20702, ErrorMessage : lock table bind changed

2023-12-24 00:03:48 ERROR TransExecutor:263 -
ErrorTime : 2023-12-24 00:03:48
RequestTime : 2023-12-24 00:03:33.608
TxnName : update_pk
Statement : UPDATE sbtest1 SET k=k+100 WHERE id = 84595;
ErrorCode : 20400
ErrorMessage : invalid state local lock table closed
Expected : false

2023-12-24 00:03:49 ERROR TransExecutor:129 -
ErrorTime : 2023-12-24 00:03:49
RequestTime : 2023-12-24 00:03:17.464
TxnName : insert_delete
Statement : delete from sbtest3 where id = 4938281;
ErrorCode : 20701
ErrorMessage : deadlock detected
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
Expected : false

2023-12-24 00:03:49 ERROR TransExecutor:129 -
ErrorTime : 2023-12-24 00:03:49
RequestTime : 2023-12-24 00:03:17.460
TxnName : insert_delete
Statement : delete from sbtest1 where id = 4937708;
ErrorCode : 20701
ErrorMessage : deadlock detected
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
Expected : false

### Expected Behavior

the error "deadlock detected" should not be expected in these cases.

### Steps to Reproduce

```Markdown
1.run a mo cluster with the config in this issue 
2.run stability test
  run tpch 100 loop queries (in a tenant)
  run tpcc 10-10 benchmark (int a tenant)
  run sysbench mixed cases with 25 threads (in a tenant)
  run another sysbench mixed cases with 25 threads(in a tenant)
```

**Comment by heni02** [comment]
log:
<img width="1407" alt="企业微信截图_0a8cc59c-49a3-480c-9147-0295a8fb6093" src="https://github.com/matrixorigin/matrixone/assets/113406637/56350022-ca8d-4700-b4de-4f32707ccd07">
http://10.222.6.1/explore?panes=%7B%22kRo%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22mo-stability-regression-20231223%5C%22%7D%20%7C%3D%20%60deadlock%20detected%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221703318400000%22,%22to%22:%221703405050000%22%7D%7D%7D&schemaVersion=1&orgId=1
[Explore-logs-2023-12-25 17_30_49.txt](https://github.com/matrixorigin/matrixone/files/13765368/Explore-logs-2023-12-25.17_30_49.txt)

**Comment by zhangxu19830126** [comment]
fixed by #13886

**Comment by aressu1985** [comment]
fixed
